### PR TITLE
ARC: Add MWDT support into qemu_arc_hs platform

### PR DESCRIPTION
**Pull request by kokas-a**
Add Metaware toolchain into qemu_arc_hs* platforms

### CODE REVIEW

**Review by abrodkin — DISMISSED** [review]

File: soc/arc/snps_qemu/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,18 @@
...
+if(COMPILER STREQUAL gcc)
+	zephyr_compile_options(-mcpu=${GCC_M_CPU})
+
+
+	if(CONFIG_ISA_ARCV2)
+	  zephyr_compile_options(-mno-sdata)
+	endif()
+else()
+
+	zephyr_compile_options( -arcv2hs -core2 -Xatomic
```
Comment (ruuddw) [inline soc/arc/snps_qemu/CMakeLists.txt:12]: `-arcv2hs` and `-Xatomic` don't seem right for arc_qemu_em - I think some more "if's" will be needed, see e.g. emsdp.

**Review by ruuddw — CHANGES_REQUESTED** [review]

File: soc/arc/snps_qemu/CMakeLists.txt (hunk 1 above)
Comment (evgeniy-paltsev) [inline soc/arc/snps_qemu/CMakeLists.txt:12]: This shoulde be something like `zephyr_compile_options_ifdef(CONFIG_SOC_QEMI_ARC_HS ...` as currently we only add options for one configuration. Same for zephyr_ld_options, you can use https://github.com/zephyrproject-rtos/zephyr/blob/dca33a126d1765cb647c10eabeea36eafcbd0f92/soc/arc/snps_nsim/CMakeLists.txt#L52 as reference.

**Review by evgeniy-paltsev — CHANGES_REQUESTED** [review]

File: soc/arc/snps_qemu/CMakeLists.txt (hunk 1 above)
Comment (kokas-a) [reply soc/arc/snps_qemu/CMakeLists.txt]: Fixed

Comment (kokas-a) [reply soc/arc/snps_qemu/CMakeLists.txt]: Done

**Review by evgeniy-paltsev — DISMISSED** [review]

File: soc/arc/snps_qemu/CMakeLists.txt [hunk 2]
```diff
@@ -1,5 +1,23 @@
 # SPDX-License-Identifier: Apache-2.0
 
-if(CONFIG_ISA_ARCV2)
-  zephyr_compile_options(-mno-sdata)
+if(COMPILER STREQUAL gcc)
+	zephyr_compile_options(-mcpu=${GCC_M_CPU})
+
+
+	if(CONFIG_ISA_ARCV2)
+	  zephyr_compile_options(-mno-sdata)
```
Comment (ruuddw) [inline soc/arc/snps_qemu/CMakeLists.txt:8]: Let's use `zephyr_compile_options_ifdef` here as well then.

File: soc/arc/snps_qemu/CMakeLists.txt [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+
+	zephyr_compile_options_ifdef(CONFIG_SOC_QEMU_ARC_HS -arcv2hs -core2 -Xatomic
+			       -Xunaligned -Xcode_density -Xswap -Xbitscan
+			       -Xmpy_option=qmpyh -Xshift_assist -Xbarrel_shifter
+			       -Xtimer0 -Xtimer1)
+
+	zephyr_ld_option_ifdef(CONFIG_SOC_QEMU_ARC_HS -Hlib=hs38_full)
+
+	if(NOT CONFIG_SOC_QEMU_ARC_HS)
+		message(WARNING "QEMU ARC platforms other than HS are not supported yet with MW toolchain")
```
Comment (ruuddw) [inline soc/arc/snps_qemu/CMakeLists.txt:20]: Any reason for not supporting EM in one go?

**Review by ruuddw — CHANGES_REQUESTED** [review]

File: soc/arc/snps_qemu/CMakeLists.txt (hunk 3 above)
Comment (kokas-a) [reply soc/arc/snps_qemu/CMakeLists.txt:20]: EM platform has never been tested yet. The idea was to add support step-by-step

File: soc/arc/snps_qemu/CMakeLists.txt (hunk 2 above)
Comment (kokas-a) [reply soc/arc/snps_qemu/CMakeLists.txt]: Done

**Review by evgeniy-paltsev — APPROVED** [review]

**Review by abrodkin — APPROVED** [review]

**Review by ruuddw — APPROVED** [review]
Please create an issue to not forget about adding EM support as well.

**Pull request merged**